app/src/main/kotlin/at/bitfire/davdroid/repository/AccountRepository.kt

```diff
@@ -170,6 +172,28 @@ class AccountRepository @Inject constructor(
 
     fun getAll(): Array<Account> = accountManager.getAccountsByType(accountType)
 
+    /**
+     * Finds and returns the main account of the given address book's account
+     *
+     * @param account the address book account to find the main account for
+     *
+     * @return the associated main account, `null` if none can be found (e.g. when main account has been deleted)
+     *
+     * @throws IllegalArgumentException when [account] is not an address book account
+     */
+    fun mainAccount(account: Account): Account? =
+        if (account.type == context.getString(R.string.account_type_address_book)) {
+            val manager = AccountManager.get(context)
+            manager.getUserData(account, USER_DATA_COLLECTION_ID)?.toLongOrNull()?.let { collectionId ->
+                collectionRepository.get(collectionId)?.let { collection ->
+                    serviceRepository.get(collection.serviceId)?.let { service ->
+                        Account(service.accountName, accountType)
+                    }
+                }
+            }
+        } else
+            throw IllegalArgumentException("$account is not an address book account")
+
```

Comment:
I think we can drop this completely. Current remaining important usages:

1. `AccountSettings`: must not be used with other accounts than DAVx5 ("main") accounts.
2. `SyncManager`: we can get the account from `collection` instead of `localCollection`.

Reply:
Maybe.  For the `SyncManager`. I think that works, but for `AccountSettings`: the address books still have the group method stored in the main account settings. See `AddressBookSyncer`. I am not sure how to deal with that. But, yes. Let's have a call.